Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -2646,21 +2648,27 @@ void webkitWebViewSetIcon(WebKitWebView* webView, const LinkIcon& icon, API::Dat
 }
 #endif
 
-RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, const WindowFeatures& windowFeatures, WebKitNavigationAction* navigationAction)
+RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, Ref<API::PageConfiguration>&& configuration, WindowFeatures&& windowFeatures, WebKitNavigationAction* navigationAction)
 {
+    RefPtr openerProcess = configuration->openerProcess();
+
+    ASSERT(!webView->priv->configurationForNextRelatedView);
+    SetForScope configurationScope(webView->priv->configurationForNextRelatedView, WTFMove(configuration));
+
     WebKitWebView* newWebView;
     g_signal_emit(webView, signals[CREATE], 0, navigationAction, &newWebView);
     if (!newWebView)
         return nullptr;
 
-    if (&getPage(webView).process() != &getPage(newWebView).process()) {
+    Ref newPage = getPage(newWebView);
+    if (&getPage(webView) != newPage->configuration().relatedPage() || openerProcess != newPage->configuration().openerProcess()) {
         g_warning("WebKitWebView returned by WebKitWebView::create signal was not created with the related WebKitWebView");
         return nullptr;
     }
 
     webkitWindowPropertiesUpdateFromWebWindowFeatures(newWebView->priv->windowProperties.get(), windowFeatures);
 
-    return &getPage(newWebView);
+    return newPage;
```

Comment:
Since this is converting a `Ref` to a `RefPtr`, I think we might save one reference count churn by writing `WTFMove`, even though that would be wrong if the return type was `Ref`.

Reply:
I'll try it.

Reply:
Works.

Reply:
Well, it worked for me only because I tested with Clang, which doesn't have -Wredundant-move or is more lenient about it. GCC doesn't like it. I'll revert this before landing.

Reply:
Um, that must mean that it uses the rvalue constructor automatically (otherwise it wouldn't be "redundant"). It must be the guaranteed copy elision, so I think there should be no refcount churn.

Reply:
Some day I will learn these rules.